Tweak wording regarding command names and bulk overwrites.

Mehgugs opened the pull request:
Changes the wording surrounding command names to match observed behaviour
w.r.t upper case characters with case folding rules that produce other
upper case characters, and characters for which case makes no sense.

Changes the wording so that bulk overwrites cannot be confused for a bulk
itemwise overwrite, which they are not.

Mehgugs committed:
Tweak wording regarding command names and bulk overwrites.

Changes the wording surrounding command names to match observed behaviour
w.r.t upper case characters with case folding rules that produce other
upper case characters, and characters for which case makes no sense.

Changes the wording so that bulk overwrites cannot be confused for a bulk
itemwise overwrite, which they are not.

Nihlus commented on docs/interactions/Application_Commands.md:
```diff
@@ -4,8 +4,9 @@ Application commands are commands that an application can register to Discord. T
-> `CHAT_INPUT` command names must be all lowercase matching `^[\w-]{1,32}$`. `USER` and `MESSAGE` commands may be mixed case and can include spaces.
+###### Application Command Naming
+
+`CHAT_INPUT` command names and command option names must be a combination of up to 32 letters, digits, the underscore charater '_' or the hyphen character '-'. If there is a lowercase variant of any of the letters you must use those, characters with no lowercase variants and/or uncased letters are still allowed. `USER` and `MESSAGE` commands may be mixed case and can include spaces.
```
Personally, I find this wording is less readable than the existing one. The regular expression is, though not perfectly accurate, far more expressive than the suggested change.

Mehgugs replied:
The reason I have removed the regex is because it's ambiguous without more information. Regexes are implementation dependent and not unicode aware by default in some engines.
 
This description is more accurate in these cases because, by providing the category of character or the character itself, people can construct matching rules to suit their system. I for one use regex alternatives and this documentation is much more informative. 

Secondly "lowercase matching this regex" is wrong. You can use uppercase characters as long as they have no lower case equivalent (i.e after case folding they do not produce a lower case letter), you can also use any letter which is not classed as a cased letter. I've made this change to be sure there's no interpretation necessary regarding regex engines and to describe some of the nuances of the actual system.

I'm happy to provide more explicit unicode information if you think that would be more helpful.

SinisterRectus replied:
> underscore charater

Typo on `character`.

> If there is a lowercase variant of any of the letters you must use those, characters with no lowercase variants and/or uncased letters are still allowed.

Change `,` to `;` or split to a new sentence. Maybe a `,` after `letters`.

Mehgugs committed:
Spelling and grammar fixes.

Mehgugs committed:
Character class with the unicode flag should be explicit enough.

On reflection, the regex should be clear to most people if the unicode
flag is mentioned explicitly.

msciotti commented on docs/interactions/Application_Commands.md:
```diff
@@ -4,8 +4,9 @@ Application commands are commands that an application can register to Discord. T
 
-> info
-> `CHAT_INPUT` command names must be all lowercase matching `^[\w-]{1,32}$`. `USER` and `MESSAGE` commands may be mixed case and can include spaces.
+###### Application Command Naming
```
This subheading format is a bit funky because we tend to use 6 hashes just for table headers, but putting it any more top-level would mean that the rest of the tables would get linked to the naming subheading, which would be weird.

The section could be moved somewhere else, but it makes sense here.

All in all, fine, just thinking out loud.
